serde-generate/runtime/golang/src/lcs/serializer.go

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) Facebook, Inc. and its affiliates
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+package lcs
+
+import (
+	"bytes"
+	"errors"
+
+	"github.com/facebookincubator/serde-reflection/serde"
+)
+
+// Serializer implements `serde.Serializer` interface for serializing LCS bytes
+type Serializer struct {
+	buf bytes.Buffer
+}
+
+// NewSerializer creates a new `serde.Serializer`
+func NewSerializer() serde.Serializer {
+	return new(Serializer)
+}
+
+func (s *Serializer) GetBytes() []byte {
+	return s.buf.Bytes()
+}
+
+func (s *Serializer) SerializeLen(value int) error {
+	if value < 0 {
+		return errors.New("length must >= 0")
+	}
+	s.serializeU32AsUleb128(uint32(value))
+	return nil
+}
+
+func (s *Serializer) SerializeBytes(value []byte) error {
+	s.SerializeLen(len(value))
+	s.buf.Write(value)
+	return nil
```

Comment:
Sadly, each of those `s.buf.Write(value)` calls can fail in theory. Not sure if it's worth the trouble. (Looking at the docs, it seems like it won't panic in case of OOM)

Reply:
Write appends the contents of p to the buffer, growing the buffer as needed. The return value n is the length of p; err is always nil. If the buffer becomes too large, Write will panic with ErrTooLarge.

Intend to return nil and tell user, you can ignore the error returned, in case you really want to cover all branches (test error case)

Reply:
by the way, I prefer test all cases including errors.

Reply:
The following code probably tells everything:

``` golang
func (b *Buffer) Write(p []byte) (n int, err error) {
	b.lastRead = opInvalid
	m, ok := b.tryGrowByReslice(len(p))
	if !ok {
		m = b.grow(len(p))
	}
	return copy(b.buf[m:], p), nil
}
```

Reply:
indeed